src/extensions/score_metamodel/checks/id_contains_feature.py

```diff
@@ -45,7 +46,12 @@ def id_contains_feature(app: Sphinx, need: NeedsInfoType, log: CheckLogger):
     # NOTE: This does not match the process requirements
     docname = dir_docname if dir_docname else need.get("docname", "")
 
-    if feature not in docname:
+    found = False
+    for featurepart in featureparts:
+        if featurepart in docname:
+            found = True
```

Comment:
A suggestion that avoids a for loop. Not sure if this is more readable, but would be faster. 
I'm okay with either. @AlexanderLanin what's your opinion? 

```suggestion
found = any(featurepart.lower() in docname.lower() for featurepart in featureparts)
```
We should  make it case insensitive, though.